frameworks/zcomet.zsh

```diff
@@ -0,0 +1,45 @@
+() {
+  local -r home_dir=${1}
+
+  git clone https://github.com/agkozak/zcomet.git $home_dir
+
+# add modules to .zshrc
+  >| ${home_dir}/.zshrc <<\END
+source ${HOME}/zcomet.zsh
+zcomet load zimfw/environment
+zcomet load zimfw/git
+zcomet load zimfw/input
+zcomet load zimfw/termtitle
+zcomet load zimfw/utility
+zcomet load zimfw/duration-info
```

Comment:
zimfw/git has both a init.zsh to be sourced and a functions subdir to be added to the fpath (plus autoloading git-alias-lookup git-branch-current git-branch-delete-interactive git-dir git-ignore-add git-root git-stash-clear-interactive git-stash-recover git-submodule-move git-submodule-remove which you're already doing).

zimfw/utility also has both a init.zsh to be sourced and a functions subdir to be added to the fpath (plus autoloading mkcd mkpw which you're already doing).

zimfw/duration-info also has both a init.zsh to be sourced and a functions subdir to be added to the fpath (plus autoloading duration-info-precmd duration-info-preexec which you're already doing).

Reply:
FYI, we use [prezto-format modules](https://github.com/zimfw/zimfw/releases/tag/v1.4.3) in Zim.

Reply:
Thanks, @ericbn! I think that all of that is actually happening; I put a little debugging code in there, and these files are being sourced by the `_zcomet_load` function:

    /path/to/...
    zimfw/environment/init.zsh
    zimfw/git/init.zsh
    zimfw/input/init.zsh
    zimfw/termtitle/init.zsh
    zimfw/utility/init.zsh
    zimfw/duration-info/init.zsh
    zimfw/asciiship/asciiship.zsh-theme
    zsh-autosuggestions/zsh-autosuggestions.plugin.zsh
    zsh-users/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh
    zsh-users/zsh-history-substring-search/zsh-history-substring-search.plugin.zsh

The `fpath` contains

    /path/to/...
    zsh-users/zsh-history-substring-search
    zsh-users/zsh-syntax-highlighting
    zsh-users/zsh-autosuggestions
    zsh-users/zsh-completions/src
    zimfw/asciiship
    zimfw/git-info/functions
    zimfw/duration-info/functions
    zimfw/utility/functions
    zimfw/termtitle
    zimfw/input
    zimfw/git/functions
    zimfw/environment

Let me know if there's something missing.

I think what may be throwing you off is that I use the `zcomet fpath` command with an explicit `functions` or `src` directory in cases where there's no file to source. In the case of `zimfw/git-info`, `zcomet load` will actually add the `functions/` subdirectory to the `fpath`, and if you think it's more educational to do it that way, I'll change that. For `/zsh-completions/src` I think `zcomet fpath` is called for -- it's an atypical place to keep functions.